pyro/ops/integrator.py

```diff
@@ -35,7 +36,7 @@ def velocity_verlet(z, r, potential_fn, step_size, num_steps=1):
             r_next[site_name] = r_next[site_name] + 0.5 * step_size * (-grads[site_name])
     z_next = {key: Variable(val) for key, val in z_next.items()}
     r_next = {key: Variable(val) for key, val in r_next.items()}
-    return z_next, r_next
+    return z_next, r_next, grads
```

Comment:
nit: Rather than changing the interface of this integrator, I would suggest creating another method called `single_step_velocity_verlet` and call that from here. This ensures that we can use either of the methods without having to change the return type for the default method. 

For instance, we might want to return the computed potential energy value for `z_next` too in `single_step_velocity_verlet` so as to not have to recompute this using `self._energy`, and it will be cumbersome to change the default method's return type again.

```
    z_next = {key: val.data.clone() for key, val in z.items()}
    r_next = {key: val.data.clone() for key, val in r.items()}
    grads = _grad(potential_fn, z_next)
    for _ in range(num_steps):
        z_next, r_next, grads = single_step_velocity_verlet(z_next, r_next, grads)
    z_next = {key: Variable(val) for key, val in z_next.items()}
    r_next = {key: Variable(val) for key, val in r_next.items()}
    return z_next, r_next
```